AsyncDisplayKit/ASNetworkImageNode.mm

```diff
@@ -238,7 +238,11 @@ - (void)_lazilyLoadImageIfNecessary
                           callbackQueue:dispatch_get_main_queue()
                              completion:cacheCompletion];
       } else {
-        [self _downloadImageWithCompletion:finished];
+          // NSURLSessionDownloadTask will do file I/O to create a temp directory. If called on the main thread this
+          // will cause significant performance issues.
+          dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+              [self _downloadImageWithCompletion:finished];
+          });
```

Comment:
Super simple change, but this completely removes the issue with `NSURLSessionDownloadTask` that we found. Do we need a "smarter" threading system or is dispatching to the default queue sufficient?